### PR TITLE
Update TCP to SpringBoot/SpringCloud 2.0.x

### LINKED ISSUE #4: Update TCP to SpringBoot/SpringCloud 2.0.x

### PR DESCRIPTION
**Pull request by tzolov**
- set default output content-type in post-processors
 - replace config. properties inheritance by composition
 - Resolves missing documentation generation

Resolves #4

### CODE REVIEW

File: spring-cloud-starter-stream-processor-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/processor/TcpClientProcessorProperties.java [hunk 1]
```diff
@@ -90,11 +90,11 @@ public void setHost(String host) {
 	}
 
 	@NotNull
-	public Encoding getEncoder() {
+	public TcpConnectionFactoryProperties.Encoding getEncoder() {
```
Comment (artembilan) [inline spring-cloud-starter-stream-processor-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/processor/TcpClientProcessorProperties.java]: Looks like this `Encoding` is on wrong level.
Let's pull it to the top level class!

File: spring-cloud-starter-stream-processor-tcp-client/src/main/resources/META-INF/spring.factories [hunk 2]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.env.EnvironmentPostProcessor=\
+  org.springframework.cloud.stream.app.tcp.ContentTypeEnvironmentPostProcessor
```
Comment (artembilan) [inline spring-cloud-starter-stream-processor-tcp-client/src/main/resources/META-INF/spring.factories]: New line in the end of each file.
You can configure IDEA to do that for you: https://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file

File: spring-cloud-starter-stream-source-tcp-client/src/main/resources/META-INF/spring.factories [hunk 3]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.env.EnvironmentPostProcessor=\
+  org.springframework.cloud.stream.app.tcp.ContentTypeEnvironmentPostProcessor
```
Comment (artembilan) [inline spring-cloud-starter-stream-source-tcp-client/src/main/resources/META-INF/spring.factories]: DITTO

File: spring-cloud-starter-stream-source-tcp/src/main/resources/META-INF/spring.factories [hunk 4]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.env.EnvironmentPostProcessor=\
+  org.springframework.cloud.stream.app.tcp.ContentTypeEnvironmentPostProcessor
```
Comment (artembilan) [inline spring-cloud-starter-stream-source-tcp/src/main/resources/META-INF/spring.factories]: DITTO

File: tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/ContentTypeEnvironmentPostProcessor.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+
+/**
+ * An {@EnvironmentPostProcessor} to set the {@code spring.cloud.stream.bindings.output.contentType}
+ * property to {@code application/octet-stream} if it has not been set already.
+ *
+ * @author Chris Schaefer
+ */
+public class ContentTypeEnvironmentPostProcessor implements EnvironmentPostProcessor {
```
Comment (artembilan) [inline tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/ContentTypeEnvironmentPostProcessor.java]: This class must be replaced by the `ContentTypeEnvironmentPostProcessor` from the `core`.
See https://github.com/spring-cloud-stream-app-starters/scriptable-transform/commit/243da380d33229f11fdf18f9caf17c2b5f38a5b1

File: tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/TcpConnectionFactoryProperties.java [hunk 6]
```diff
@@ -16,13 +16,17 @@
...
 /**
  * Base class for TCP connection factory properties.
  *
  * @author Eric Bottard
  * @author Gary Russell
+ * @author Christian Tzolov
  */
-public abstract class AbstractTcpConnectionFactoryProperties {
+@ConfigurationProperties("tcp")
+final public class TcpConnectionFactoryProperties {
```
Comment (artembilan) [inline tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/TcpConnectionFactoryProperties.java]: Nit: I'd prefer `public final` 😄

**Review by artembilan — CHANGES_REQUESTED** [review]
That's all.

Consider also add your name to all the affected classes and update their Copyright to the current year.

Also, please, take a look into the issues https://github.com/spring-cloud-stream-app-starters/tcp/issues and mention all those you are fixing here.

Thanks

File: spring-cloud-starter-stream-processor-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/processor/TcpClientProcessorProperties.java (hunk 1 above)
Comment (tzolov) [reply spring-cloud-starter-stream-processor-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/processor/TcpClientProcessorProperties.java]: +1

File: spring-cloud-starter-stream-processor-tcp-client/src/main/resources/META-INF/spring.factories (hunk 2 above)
Comment (tzolov) [reply spring-cloud-starter-stream-processor-tcp-client/src/main/resources/META-INF/spring.factories]: +1

File: spring-cloud-starter-stream-source-tcp-client/src/main/resources/META-INF/spring.factories (hunk 3 above)
Comment (tzolov) [reply spring-cloud-starter-stream-source-tcp-client/src/main/resources/META-INF/spring.factories]: +1

File: spring-cloud-starter-stream-source-tcp/src/main/resources/META-INF/spring.factories (hunk 4 above)
Comment (tzolov) [reply spring-cloud-starter-stream-source-tcp/src/main/resources/META-INF/spring.factories]: +1

File: tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/TcpConnectionFactoryProperties.java (hunk 6 above)
Comment (tzolov) [reply tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/TcpConnectionFactoryProperties.java]: +1

Comment (tzolov) [reply tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/TcpConnectionFactoryProperties.java]: +1

File: tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/ContentTypeEnvironmentPostProcessor.java (hunk 5 above)
Comment (tzolov) [reply tcp-app-starters-common/src/main/java/org/springframework/cloud/stream/app/tcp/ContentTypeEnvironmentPostProcessor.java]: +1

**Review by artembilan — APPROVED** [review]
@tzolov ,

Cool work, indeed!

Would you mind to reconsider this my previous comment though?

> Consider also add your name to all the affected classes and update their Copyright to the current year.

> Also, please, take a look into the issues https://github.com/spring-cloud-stream-app-starters/tcp/issues and mention all those you are fixing here.

I don't mind to merge and polish a bit, but it would be better if you revise issues and mention those here which you are fixing. Just to be sure that we are on the same page!

Thanks

**Comment by tzolov** [comment]
@artembilan thanks for the reminder! I've updated the authors and dates docs and mentioned the . 

fyi, first I thought that those changes are "cosmetic" and i don't have to add my name to the authors list.

**Pull request merged**